src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.metrics;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+
+public class MetricsEmitter {
+    protected ScheduledFuture<?> future;
+    private final ScheduledExecutorService ses;
+    private static final int PERIODIC_AGGREGATE_INTERVAL = 3_600;
+    private final ClientDeviceAuthMetrics metrics;
+
+    /**
+     * Constructor for the MetricsEmitter.
+     *
+     * @param ses     {@link ScheduledExecutorService}
+     * @param metrics {@link ClientDeviceAuthMetrics}
+     */
+    @Inject
+    public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metrics) {
+        this.ses = ses;
+        this.metrics = metrics;
+    }
+
+    /**
+     * Emit metrics once an hour.
+     */
+    public void emit() {
+        future = ses.scheduleWithFixedDelay(metrics::emitMetrics, 0, PERIODIC_AGGREGATE_INTERVAL,
```

Comment:
we should protect against multiple backing tasks running at the same time--typically we cancel the existing task

Reply:
Do we want initial delay of zero? Will we have any metrics to emit then?

Reply:
We were following this format: [https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/e720757c1b0920ee771a61b73a2bcb1554bcbc33/src/main/java/com/aws/greengrass/telemetry/TelemetryAgent.java#L190](url) , which is why I went with the initial delay of 0